shoes-swt/lib/shoes/swt/gradient.rb

```diff
@@ -44,13 +44,8 @@ def apply_as_stroke(gc, dsl)
       private
 
       def extract_dimensions(dsl)
-        if dsl.is_a?(Star)
-          left = dsl.element_left - dsl.element_width / 2.0
-          top  = dsl.element_top - dsl.element_height / 2.0
-        else
-          left = dsl.element_left
-          top  = dsl.element_top
-        end
+        left = dsl.element_left
+        top  = dsl.element_top
```

Comment:
Interesting that this had that effect... don't feel like I understand why, but I haven't gotten familiar with the gradient code.

Noticed too that I think we could just unroll the `extract_dimensions` method now that it's basically just using the `dsl.element_*` accessors with no real additional computation. Saves an array allocation :wink: